test/13-device-config.spec.ts

```diff
@@ -0,0 +1,514 @@
+import * as Bluebird from 'bluebird';
+import { child_process, fs } from 'mz';
+import * as Sinon from 'sinon';
+
+import { ExtlinuxConfigBackend, RPiConfigBackend } from '../src/config/backend';
+import { DeviceConfig } from '../src/device-config';
+import * as fsUtils from '../src/lib/fs-utils';
+import { expect } from './lib/chai-config';
+
+import prepare = require('./lib/prepare');
+
+const extlinuxBackend = new ExtlinuxConfigBackend();
+const rpiConfigBackend = new RPiConfigBackend();
+
+describe('DeviceConfig', function() {
+	before(function() {
+		prepare();
+		this.fakeDB = {};
+		this.fakeConfig = {
+			get(key: string) {
+				return Bluebird.Promise.try(function() {
+					if (key === 'deviceType') {
+						return 'raspberrypi3';
+					} else {
+						throw new Error('Unknown fake config key');
+					}
+				});
+			},
+		};
+		this.fakeLogger = {
+			logSystemMessage: Sinon.spy(),
+		};
+		return (this.deviceConfig = new DeviceConfig({
+			logger: this.fakeLogger,
+			db: this.fakeDB,
+			config: this.fakeConfig,
+		}));
+	});
+
+	// Test that the format for special values like initramfs and array variables is parsed correctly
+	it('allows getting boot config with getBootConfig', function() {
+		Sinon.stub(fs, 'readFile').resolves(`\
+initramfs initramf.gz 0x00800000\n\
+dtparam=i2c=on\n\
+dtparam=audio=on\n\
+dtoverlay=ads7846\n\
+dtoverlay=lirc-rpi,gpio_out_pin=17,gpio_in_pin=13\n\
+foobar=baz\n\
+		`);
+		return this.deviceConfig
+			.getBootConfig(rpiConfigBackend)
+			.then(function(conf: any) {
+				(fs.readFile as Sinon.SinonStub).restore();
+				return expect(conf).to.deep.equal({
+					HOST_CONFIG_initramfs: 'initramf.gz 0x00800000',
+					HOST_CONFIG_dtparam: '"i2c=on","audio=on"',
+					HOST_CONFIG_dtoverlay:
+						'"ads7846","lirc-rpi,gpio_out_pin=17,gpio_in_pin=13"',
+					HOST_CONFIG_foobar: 'baz',
+				});
+			});
+	});
+
+	it('properly reads a real config.txt file', function() {
+		Sinon.stub(fs, 'readFile').resolves(`\
+dtparam=i2c_arm=on\n\
+dtparam=spi=on\n\
+dtparam=audio=on\n\
+enable_uart=1\n\
+disable_splash=1\n\
+avoid_warnings=1\n\
+gpu_mem=16\n\
+				`);
```

Comment:
The only way to format this so it is indented would require doing a bunch of string concats and have those individual strings spaced with tabs. If we add tabs to this text directly it will break how deviceConfig interprets the values.

Reply:
You can use `stripIndent` from `commonTags` - I always meant to add this. It is technically out of the scope for this PR so I'm not super worried.

Reply:
When I added stripIndent the tests would fail. I tried troubleshooting it for awhile but can't figure out how adding that is causing the strings not to match what it was before..